Remove `new Buffer` call in flow-metrics

![image](https://user-images.githubusercontent.com/128755/37316205-2a75317c-2634-11e8-9f45-2bd33c244602.png)

cc @deeptibaghel

here's the <kbd>$ npm run lint</kbd> output if you add **eslint-plugin-node**/recommended config:

```sh
➜  fxa-content-server git:(5a1dfe90) ✗ npm run lint

> fxa-content-server@1.107.4 lint /Users/pdehaan/dev/github/mozilla/fxa-content-server_pd/fxa-content-server
> eslint app server tests --cache


/Users/pdehaan/dev/github/mozilla/fxa-content-server_pd/fxa-content-server/server/lib/legal-templates.js
  35:7  error  'fs.exists' was deprecated since v4. Use 'fs.stat()' or 'fs.access()' instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla/fxa-content-server_pd/fxa-content-server/tests/server/flow-metrics.js
  24:14  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  26:15  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

✖ 3 problems (3 errors, 0 warnings)


npm ERR! Darwin 17.4.0
npm ERR! argv "/Users/pdehaan/.nvm/versions/node/v6.13.1/bin/node" "/Users/pdehaan/.nvm/versions/node/v6.13.1/bin/npm" "run" "lint"
npm ERR! node v6.13.1
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! fxa-content-server@1.107.4 lint: `eslint app server tests --cache`
npm ERR! Exit status 1
```

... And here's my tweaked .eslintrc config file:

```yaml
plugins:
  - fxa
  - node
  - sorting

extends:
  - plugin:fxa/client
  - plugin:node/recommended

# Required for `import( )` webpack feature
parser: babel-eslint

root: true

rules:
  node/no-missing-require: off
  node/no-unpublished-require: off
  node/no-unsupported-features: [error, {ignores: ["modules"]}]
  node/shebang: off
  sorting/sort-object-props: [error, {ignoreCase: true, ignoreMethods: true, ignoreObjectsWithMethods: true, ignorePrivate: true}]

  brace-style: error
  handle-callback-err: off
  key-spacing: [error, {beforeColon: false, afterColon: true}]
  keyword-spacing: error
  linebreak-style: [error, unix]
  max-len: [error, 160]
  new-cap: [error, {newIsCap: true, capIsNew: false}]
  no-console: off
  no-multi-str: error
  no-process-exit: off
  no-useless-escape: off
  space-infix-ops: error
  space-unary-ops: [error, {nonwords: false, overrides: {"~": true, "!": true}}]
  strict: off
```

--- a/tests/server/flow-metrics.js
+++ b/tests/server/flow-metrics.js
@@ -21,9 +21,9 @@ suite.beforeEach = function () {
   });
   sinon.stub(crypto, 'randomBytes').callsFake(function (size) {
     if (mockRandomBytes) {
-      return new Buffer(mockRandomBytes);
+      return Buffer.from(mockRandomBytes);
     } else {
-      var b = new Buffer(size);
+      var b = Buffer.alloc(size);
       b.fill(0);
       return b;
     }

